numpy/lib/type_check.py

```diff
@@ -262,6 +262,10 @@ def isreal(x):
     out : ndarray, bool
         Boolean array of same shape as `x`.
 
+    Notes
+    -----
+    Returns all True for non-complex array.
+
```

Comment:
IMO this statement isn't specific and doesn't clarify anything, so I'd just as soon remove it. Alternatively, the strange behavior for certain dtypes could be highlighted:

```suggestion
    Notes
    -----
    `isreal` may behave unexpectedly for string or object arrays (see examples)
```